### PR TITLE
Added Register Functionality to the Events controller.

### PR DESCRIPTION
**Pull request by XennMaestro**

### CODE REVIEW

File: app/Http/Controllers/Events.php [hunk 1]
```diff
@@ -139,4 +141,33 @@ public function destroy($id) {
...
+        if (count($eventUser) > 0) {
+            return response("userId {$userId} already registered for event id {$id}.", 404);
+        }
+
+        $userEvent = new UserEvent;
+        $userEvent->userID = $userId;
+        $userEvent->eventID = $id;
+        $userEvent->save();
+
+        return response()->json(['User Resistered' => $userEvent->id]);
```
Comment (brandonson) [inline app/Http/Controllers/Events.php]: I'd just do `['id' => $userEvent->id]`, spaces make parsing the response more difficult.

**Comment by brandonson** [comment]
You need to add a check in there to see whether the user is actually owned by the current account.  I'm also not sure how I feel about the URL structure - realistically you should probably see if you can make it take list of user IDs instead.  So, something like this as the request body:

```
{
  "attendees": [1,3,6]
}
```

That way the frontend doesn't need to hit the endpoint multiple times to register a group of people.

**Comment by XennMaestro** [comment]
@brandonson Yea ill update the single register function for now. I will work on creating a secondary group register function. I think @nikkimartin  wanted to finish the events page today so i want to push some of the changes that i have made.

**Comment by nikkimartin** [comment]
I don't need the stuff I mentioned to keep working, it just needs to be done at some point. Up to @brandonson if he thinks it's okay to merge.

@brandonson - just for reference, I need a different error code for no events associated to conference vs. conference doesn't exist (both just return 404 right now and I don't want to have to make another call to figure out which case it is) and an event seeder for the db.

Thanks guys!

**Comment by brandonson** [comment]
@nikkimartin Is just differentiating them by some kind of json okay?  On my phone, so hard to do an example, but something like "message" : "no_such_conference" with the 404 code.

@shehryar I'll take a more detailed look at this later today, but just a heads up that I think we will want to just merge as I believe we need you on testing ASAP.  I can always add any additional stuff later, I think I've got a pretty good handle on how you've done events.

**Comment by nikkimartin** [comment]
@brandonson: works for me. I think shehryar set it up to return a 404 in one case and 405 in the other. I don't really care, as long as it's consistent.

**Pull request merged**